webui/react/src/pages/SearchDetails.tsx

```diff
@@ -0,0 +1,222 @@
+import Pivot, { PivotProps } from 'hew/Pivot';
+import Notes from 'hew/RichTextEditor';
+import { Loadable } from 'hew/utils/loadable';
+import { string } from 'io-ts';
+import _ from 'lodash';
+import { useObservable } from 'micro-observables';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { unstable_useBlocker, useLocation, useNavigate, useParams } from 'react-router-dom';
+
+import Page, { BreadCrumbRoute } from 'components/Page';
+import { terminalRunStates } from 'constants/states';
+import usePermissions from 'hooks/usePermissions';
+import usePolling from 'hooks/usePolling';
+import { SettingsConfig, useSettings } from 'hooks/useSettings';
+import { paths } from 'routes/utils';
+import { getExperimentDetails, patchExperiment } from 'services/api';
+import workspaceStore from 'stores/workspaces';
+import { ExperimentBase, Note, ValueOf, Workspace } from 'types';
+import handleError, { ErrorLevel, ErrorType } from 'utils/error';
+import { isAborted, isNotFound } from 'utils/service';
+
+import ExperimentCodeViewer from './ExperimentDetails/ExperimentCodeViewer';
+import FlatRuns from './FlatRuns/FlatRuns';
+
+type Params = {
+  searchId: string;
+  tab?: ValueOf<typeof TabType>;
+};
+
+const TabType = {
+  Code: 'code',
+  Notes: 'notes',
+  Trials: 'trials',
+} as const;
+
+const TAB_KEYS = Object.values(TabType);
+const DEFAULT_TAB_KEY = TabType.Trials;
+
+const SearchDetails: React.FC = () => {
+  const { tab, searchId } = useParams<Params>();
+  const [experiment, setExperiment] = useState<ExperimentBase>();
+  const [pageError, setPageError] = useState<Error>();
+  const canceler = useRef<AbortController>();
+  const workspaces = Loadable.getOrElse([], useObservable(workspaceStore.workspaces));
+  const id = parseInt(searchId ?? '');
+
+  const navigate = useNavigate();
+  const location = useLocation();
+  const defaultTabKey = tab && TAB_KEYS.includes(tab) ? tab : DEFAULT_TAB_KEY;
+  const [tabKey, setTabKey] = useState(defaultTabKey);
```

Comment:
nit: this is really more of an initial tab key, not default. you also don't use the constant elsewhere, so you could just do
```
 const [tabKey, setTabKey] = useState(tab && TAB_KEYS.includes(tab) ? tab : DEFAULT_TAB_KEY);
```